packages/router/_types.ts

```diff
@@ -26,10 +26,16 @@ export interface ReactiveRouteOptions {
 
 export interface ReactiveRouteOptionsWithTransform<V, R> extends ReactiveRouteOptions {
   /**
-   * Function to transform data before return
+   * Function to transform data before return, or an object with one or both functions:
+   * `get` to transform data before returning, and `set` to transform data before setting
    */
-  transform?: ((val: V) => R) | {
-    get: (value: V) => R
-    set: (value: R) => V
-  }
+  transform?:
+    | ((val: V) => R)
+    | ({
+      get?: (value: V) => R
+      set?: (value: R) => V
+    } & (
+        { get: (value: V) => R }
+        | { set: (value: R) => V }
+      ))
```

Comment:
this is just this, right?
```suggestion
  transform?: ((val: V) => R) | Partial<{
    get: (value: V) => R
    set: (value: R) => V
  }>
```

Reply:
Not exactly. The initial version is stricter and was added to prevent passing `transform: {}`, enforcing that either `get` or `set` must be provided. But you are right, the runtime code already handles this case safely with fallbacks, so even `transform: {}` wouldn't cause unexpected behavior. Strictly speaking, we could use `Partial` as well.

Eventually, I think it comes down to priorities:

- Do we prioritize cleaner types for readability and flexibility? Then we can go with `Partial`
- Or do we enforce API clarity for end users via stricter type checks, even though the runtime already handles nonsensical parameters gracefully?

I'm open to either approach and happy to go with what aligns best with the project's style

Reply:
> Not exactly. The initial version is stricter and was added to prevent passing `transform: {}`, enforcing that either `get` or `set` must be provided

I don't think there's much point in having such a strict typecheck if it might just confuse users with (maybe) "complex" TS warnings. But thats just opionated 😄 i am fine with both 👍🏽

Reply:
In the worst case, if someone uses `transform: {}` it'll give both the default behaviors.

Reply:
I ([alexchexes](https://github.com/alexchexes)) replied to this yesterday, but my reply got a "Pending" status, and nobody can see it. Anyone knows why, and how are we supposed to do open-source when simple real-time discussion is impossible here?

Reply:
> I ([alexchexes](https://github.com/alexchexes)) replied to this yesterday, but my reply got a "Pending" status, and nobody can see it. Anyone knows why, and how are we supposed to do open-source when simple real-time discussion is impossible here?
> 
> P.S. Now all my replies from the initial account are blocked with "Pending" status as well. Wth?

I think you have started a rwiew instead of adding a comment. In top right should be a blue button. If you submit the Review all comments will be published.

Reply:
Strange. I remember just adding a regular reply, just like I’m doing right now. Anyway, thanks, now it’s visible. Check it out

Reply:
since the underlying logic does handle `{}`, and the error will be cleaner, I think we should just relax the type to a `Partial`

in reality, it should actually be `GetAndSet | Get | Set` since `Partial<GetAndSet> & (Get | Set)` means if you set both, the type will actually show as one of the properties being required

Reply:
> but the error would be equally ugly

Just to clarify: the screenshots weren’t showing what happens with `transform: {}`. They were meant to compare how errors are reported when passing an invalid key like `foo` instead of `get` or `set`.  
With `Partial`, no error is raised when passing `{}` at all.  

So the trade-off is between:  

1. A cleaner type (`Partial`), which allows `{}` even though it makes no sense.  
2. A stricter type that looks more complex (including TS warnings) but prevents `{}`.  

I’m still open to whichever style is preferred here, just wanted to make sure we’re all clear on what the screenshots were illustrating

Reply:
With the proposed example, it could look like this:
```typescript
interface GetFn<V, R> {
  get: (value: V) => R
}

interface SetFn<V, R> {
  set: (value: R) => V
}

type GetterOrSetter<V, R> = GetFn<V, R> | SetFn<V, R> | (GetFn<V, R> & SetFn<V, R>)

export interface ReactiveRouteOptionsWithTransform<V, R> extends ReactiveRouteOptions {
  /**
   * Function to transform data before returning, or an object with one or both functions:
   * `get` to transform data before returning, and `set` to transform data before setting
   */
  transform?: ((value: V) => R) | GetterOrSetter<V, R>
}
```
That's better than initial option, although I'm not sure if we should really complicate things like this or just stick with `Partial`. I'm not aiming to make it "perfect," so I'll leave this decision to you guys, whoever has the "last word" :)